mining.go

```diff
@@ -117,7 +159,27 @@ func txPQByPriority(pq *txPriorityQueue, i, j int) bool {
 		return pq.items[i].feePerKB > pq.items[j].feePerKB
 	}
 	return pq.items[i].priority > pq.items[j].priority
+}
+
+// txPQByStakeAndPriority sorts a txPriorityQueue by stake priority then
+// transaction priority, and then fees per kilobyte.
+func txPQByStakeAndPriority(pq *txPriorityQueue, i, j int) bool {
+	// Sort by stake priority, continue if they're the same stake priority.
+	cmp := compareStakePriority(pq.items[i], pq.items[j])
```

Comment:
Now that the `txStakePriority` function was refactored, wouldn't this read cleaner by doing the comparison here instead of effectively repeating it in the func and checking the result here?

i.e
```Go
	iStakePriority := txStakePriority(pq.items[i].txType)
	jStakePriority := txStakePriority(pq.items[j].txType)
	if iStakePriority > jStakePriority {
		return true
	}
	if iStakePriority < jStakePriority {
		return false
	}

	...
```